main.nf

```diff
@@ -347,63 +345,100 @@ workflow {
+        fastqc_html    = FASTQC.out.html
+        fastqc_version = FASTQC.out.version
+        fastqc_zip     = FASTQC.out.zip
+    }
+
+    def bwamem2_mem_options = [:]
+
+    bwamem2_mem_options.args_bwamem2 = "-K 100000000 -M"
+    trim_galore_report = Channel.empty()
 
     if (params.trim_fastq) {
         TRIM_GALORE(input_sample)
-        result_trim_galore = TRIM_GALORE.out.report
-        BWAMEM2_MEM(TRIM_GALORE.out.trimmed_reads, bwa, fasta, fai)
-    }
-    else {
-        result_trim_galore = Channel.empty()
-        BWAMEM2_MEM(input_sample, bwa, fasta, fai)
+        BWAMEM2_MEM(TRIM_GALORE.out.trimmed_reads, bwa, fasta, fai, bwamem2_mem_options)
+        trim_galore_report = TRIM_GALORE.out.report
     }
-
-    BWAMEM2_MEM.out.groupTuple(by:[0, 1])
-        .branch {
-            single:   it[2].size() == 1
-            multiple: it[2].size() > 1
+    else BWAMEM2_MEM(input_sample, bwa, fasta, fai, bwamem2_mem_options)
+
+    results = BWAMEM2_MEM.out.map{ meta, bam, bai ->
+        patient = meta.patient
+        sample  = meta.sample
+        gender  = meta.gender
+        status  = meta.status
+        [patient, sample, gender, status, bam, bai]
+    }.groupTuple(by: [0,1])
+        .branch{
+            single:   it[4].size() == 1
+            multiple: it[4].size() > 1
         }.set { bam }
 
     bam_single = bam.single.map {
-        idPatient, idSample, idRun, bam, bai ->
-        [idPatient, idSample, bam[0], bai[0]]
+        patient, sample, gender, status, bam, bai ->
+
+        def meta = [:]
+        meta.patient = patient
+        meta.sample = sample
+        meta.gender = gender[0]
+        meta.status = status[0]
+        meta.id = sample
+
+        [meta, bam[0], bai[0]]
+    }
+
+    bam_multiple = bam.multiple.map {
+        patient, sample, gender, status, bam, bai ->
+
+        def meta = [:]
+        meta.patient = patient
+        meta.sample = sample
+        meta.gender = gender[0]
+        meta.status = status[0]
+        meta.id = sample
+
+        [meta, bam, bai]
     }
 
-    //multipleBam = multipleBam.mix(multipleBamSentieon)
+    // multipleBam = multipleBam.mix(multipleBamSentieon)
 
-    bam_mapped = bam_single.mix(MERGE_BAM_MAPPED(bam.multiple))
+    bam_mapped = bam_single.mix(MERGE_BAM_MAPPED(bam_multiple))
 
     bam_mapped.view()
 
-    if(!(params.skip_markduplicates)){
-        MARK_DUPLICATES(bam_mapped)
-        mark_duplicates_report = MARK_DUPLICATES.out.duplicates_marked_report
-        bam_duplicates_marked =  MARK_DUPLICATES.out.bam_duplicates_marked 
-    }
-    else {
-        mark_duplicates_report = Channel.empty()
-        bam_duplicates_marked = Channel.empty()
+    mark_duplicates_report = Channel.empty()
+    bam_duplicates_marked  = Channel.empty()
+
+    if (!(params.skip_markduplicates)) {
+        // MARK_DUPLICATES(bam_mapped)
+        // mark_duplicates_report = MARK_DUPLICATES.out.report
+        // bam_duplicates_marked  =  MARK_DUPLICATES.out.bam 
     }
```

Comment:
Shouldn't this go back in? Or is it not working currently?

Reply:
Yes, it's not working at the moment, and I have no idea why...

Reply:
Story of our lives with DSL2 😂

Reply:
and the error message is not helping at all...